plugins/org.eclipse.sirius.diagram.sequence.ui/src/org/eclipse/sirius/diagram/sequence/ui/tool/internal/edit/part/InteractionContainerEditPart.java

```diff
@@ -0,0 +1,240 @@
+            super.installEditPolicy(key, editPolicy);
+        }
+    }
+
+    @Override
+    protected void createDefaultEditPolicies() {
+        super.createDefaultEditPolicies();
+
+        // Handle $endBefore for launch tools.
+        installEditPolicy(org.eclipse.sirius.diagram.ui.tools.api.requests.RequestConstants.REQ_LAUNCH_TOOL, new SequenceLaunchToolEditPolicy());
+    }
+
+    @Override
+    protected NodeFigure createMainFigure() {
+        NodeFigure figure = super.createMainFigure();
+        forceCombinedFragmentDefaultSize(figure);
+        return figure;
+    }
+
+    @Override
+    protected void addDropShadow(NodeFigure figure, IFigure shape) {
+        // Remove the shadow border to avoid unwanted spacing
+        figure.setBorder(null);
+    }
+
+    @Override
+    public void refresh() {
+        if (!(getParent() instanceof InteractionContainerEditPart)) {
+            super.refresh();
+            forceCombinedFragmentDefaultSize((NodeFigure) getMainFigure());
+        }
+    }
+
+    public void setSize(Dimension newDimension) {
+        IFigure mainFigure = getMainFigure();
+        if (mainFigure instanceof DefaultSizeNodeFigure) {
+            EObject eObj = this.resolveSemanticElement();
+            if (eObj instanceof DDiagramElementContainer) {
+                DDiagramElementContainer container = (DDiagramElementContainer) eObj;
+                if (container.getOwnedStyle() instanceof FlatContainerStyle) {
+                    ((DefaultSizeNodeFigure) mainFigure).setLocation(new Point(0, 0));
+                    ((DefaultSizeNodeFigure) mainFigure).setDefaultSize(newDimension);
+                }
+            }
+        }
+    }
+
+    private void forceCombinedFragmentDefaultSize(NodeFigure figure) {
+        if (figure instanceof DefaultSizeNodeFigure) {
+            EObject eObj = this.resolveSemanticElement();
+            if (eObj instanceof DDiagramElementContainer) {
+                DDiagramElementContainer container = (DDiagramElementContainer) eObj;
+                Dimension computeInteractionContainerSize = computeInteractionContainerSize();
+                if (container.getOwnedStyle() instanceof FlatContainerStyle) {
+                    ((DefaultSizeNodeFigure) figure).setLocation(new Point(0, 0));
+                    ((DefaultSizeNodeFigure) figure).setDefaultSize(computeInteractionContainerSize);
+                }
+            }
+        }
+    }
+
+    private SequenceDiagramEditPart getSequenceDiagramEditPart() {
+        EditPart parent = this.getParent();
+        while (parent != null && !(parent instanceof SequenceDiagramEditPart)) {
+            parent = parent.getParent();
+        }
+        return (SequenceDiagramEditPart) parent;
+    }
+
+    private Dimension computeInteractionContainerSize() {
+        int computedHeight = DEFAULT_HEIGHT;
+        int computedWidth = DEFAULT_WIDTH;
+        for (EditPart childEditPart : getSequenceDiagramEditPart().getChildren()) {
```

Comment:
getSequenceDiagramEditPart() could return null

Reply:
I have not seen it even with an empty diagram but I added a check to be sure.